include/revng/Pypeline/ObjectID.h

```diff
@@ -0,0 +1,109 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <set>
+
+#include "llvm/ADT/StringRef.h"
+
+#include "revng/Model/TypeDefinition.h"
+#include "revng/Pipeline/Location.h"
+#include "revng/Pipes/Ranks.h"
+#include "revng/Support/MetaAddress.h"
+
+class ObjectID {
+private:
+  using TypeDefinitionKey = model::TypeDefinition::Key;
+  std::variant<std::monostate, MetaAddress, TypeDefinitionKey> Component;
+
+public:
+  enum Kind {
+    Root = 0,
+    Function = 1,
+    TypeDefinition = 2
+  };
+
+public:
+  ObjectID() = default;
```

Comment:
Drop

Reply:
I didn't drop it because there's now 3 constructors, which are needed for `Model.children`

Reply:
This does not apply here:
```
If a class requires a user-defined destructor, a user-defined copy constructor, or a user-defined copy assignment operator, it almost certainly requires all three.
```

In this case there's only 3 constructors, no copy/move/destructor